### ISSUE TITLE
TypeError: Can't instantiate abstract class KMeansSMOTE with abstract methods _fit_resample

### ISSUE THREAD

**Issue by StatML** [felix-last/kmeans_smote#6]
Hi, @felix-last  Thanks for sharing your great job. When I run the following codes, I met a Type Error. It seems that KMeansSMOTE didn't implement the abstract method `_fit_resample`. Can you help to fix it?

> Using TensorFlow backend.
> Class -1 has 896 instances
> Class 1 has 41 instances
> Traceback (most recent call last):
>   File "test_kmeans_smote.py", line 12, in <module>
>     smote_args={'k_neighbors': 10})
> TypeError: Can't instantiate abstract class KMeansSMOTE with abstract methods _fit_resample
> 

```
import numpy as np
from imblearn.datasets import fetch_datasets
from kmeans_smote import KMeansSMOTE

datasets = fetch_datasets(filter_data=['oil'])
X, y = datasets['oil']['data'], datasets['oil']['target']

[print('Class {} has {} instances'.format(label, count))
 for label, count in zip(*np.unique(y, return_counts=True))]

kmeans_smote = KMeansSMOTE(kmeans_args={'n_clusters': 100},
    smote_args={'k_neighbors': 10})
X_resampled, y_resampled = kmeans_smote.fit_sample(X, y)

[print('Class {} has {} instances after oversampling'.format(label, count))
 for label, count in zip(*np.unique(y_resampled, return_counts=True))]
```

**Comment by vlainic** [comment]
I have the same issue.

I tried to delete the methode, but Inget AttributeError then.

== **UPDATE** ==
I did installation of the package, and I got this:

yellowbrick 0.9.1 has requirement scikit-learn>=0.20, but you'll have scikit-learn 0.19.2 which is incompatible.
tensorpack 0.9.1 has requirement msgpack-numpy>=0.4.4.2, but you'll have msgpack-numpy 0.4.3.2 which is incompatible.
spacy 2.0.18 has requirement numpy>=1.15.0, but you'll have numpy 1.14.6 which is incompatible.
lime 0.1.1.33 has requirement matplotlib==2.1.0, but you'll have matplotlib 3.0.3 which is incompatible.
imgaug 0.2.8 has requirement numpy>=1.15.0, but you'll have numpy 1.14.6 which is incompatible.
featuretools 0.6.1 has requirement scikit-learn>=0.20.0, but you'll have scikit-learn 0.19.2 which is incompatible.
fastai 1.0.50.post1 has requirement numpy>=1.15, but you'll have numpy 1.14.6 which is incompatible.
dask-ml 0.12.0 has requirement scikit-learn>=0.20, but you'll have scikit-learn 0.19.2 which is incompatible.
cvxpy 1.0.21 has requirement numpy>=1.15, but you'll have numpy 1.14.6 which is incompatible.
Installing collected packages: scikit-learn, numpy, imbalanced-learn
  Found existing installation: scikit-learn 0.20.3
    Uninstalling scikit-learn-0.20.3:
      Successfully uninstalled scikit-learn-0.20.3
  Found existing installation: numpy 1.16.2
    Uninstalling numpy-1.16.2:
      Successfully uninstalled numpy-1.16.2
  Found existing installation: imbalanced-learn 0.5.0.dev0
    Uninstalling imbalanced-learn-0.5.0.dev0:
      Successfully uninstalled imbalanced-learn-0.5.0.dev0
Successfully installed imbalanced-learn-0.3.3 numpy-1.14.6 scikit-learn-0.19.2

**Could it be that this library is not compatible with newest Python versions?**

**Comment by felix-last** [comment]
It looks like you're using an incompatible imbalanced-learn version (0.4.x). The [API was changed in 0.4.0](https://github.com/scikit-learn-contrib/imbalanced-learn/releases/tag/0.4.0). To resolve for kmeans-smote 0.1.0, run `pip install imbalanced-learn==0.3.4`.

Even better, I just released kmeans-smote 0.1.1 to be compatible with 0.4.x (breaking compatibility with 0.3.x). **So simply `pip install --upgrade kmeans-smote` and all should be fine.**
